src/metabase/driver/sql_jdbc/common.clj

```diff
@@ -1,6 +1,39 @@
 (ns metabase.driver.sql-jdbc.common
   (:require [clojure.string :as str]))
 
+(def ^:private valid-separator-styles #{:url :comma :semicolon})
+
+(defn conn-str-with-additional-opts
+  "Adds `additional-opts` (a string) to the given `connection-string` based on the given `separator-style`. See
+  documentation for `handle-additional-options` for further details."
+  {:added "0.39.0", :arglists '([connection-string separator-style additional-opts])}
+  [connection-string separator-style additional-opts]
+  {:pre [(string? connection-string)
+         (or (nil? additional-opts) (string? additional-opts))
+         (contains? valid-separator-styles separator-style)]}
+  (str connection-string (when-not (str/blank? additional-opts)
+                           (str (case separator-style
+                                  :comma     ","
+                                  :semicolon ";"
+                                  :url       (if (str/includes? connection-string "?")
+                                               "&"
+                                               "?"))
+                                additional-opts))))
+
+(defn additional-opts->string
+  "Turns a map of `additional-opts` into a single string, based on the `separator-style`."
+  {:added "0.39.0", :arglists '([separator-style additional-opts])}
```

Comment:
Same thing, gotta update that `:added` metadata

Reply:
You don't need to specify this `:arglists` metadata here, it will get it automatically since it's a `defn`. You want to specify `:arglists` if it's a `defmulti` or plain `def`, since they don't get the metadata automatically, or if you want to `:arglists` metadata to differ from the actual args vector for the `defn`

Reply:
Both fixed in next commit.